internal/k8s/secret.go

```diff
@@ -0,0 +1,84 @@
+package k8s
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SecretClient struct {
+	k8sClient client.Client
+	log       *logr.Logger
+}
+
+func NewSecretClient(k8sClient client.Client, log *logr.Logger) (*SecretClient, error) {
+	if k8sClient == nil {
+		return nil, errors.New("k8sClient cannot be nil")
+	}
+	if log == nil {
+		return nil, errors.New("logger cannot be nil")
+	}
+	return &SecretClient{
+		k8sClient: k8sClient,
+		log:       log,
+	}, nil
+}
+
+// EnsureFinalizer idempotently adds the specified finalizer to the specified secret
+func (s *SecretClient) EnsureFinalizer(ctx context.Context, secretName, secretNamespace, finalizer string) error {
```

Comment:
Could you accept a namespacedName instead of two args? Same elsewhere

Reply:
I'm accepting a client.ObjectKey as otherwise it would require a conversion in the function